### DISCUSSION TITLE
Info about the choice to use `mpsc::sync_channel` over `mpsc::channel`

### DISCUSSION THREAD

**Discussion by GyulyVGC** [kpcyrd/sniffglue#125] [General]
Hey @kpcyrd I was looking at your code and was wondering what made you choose `sync_channel`s with capacity of 256 instead of using asynchronous `channel`s.

Did you do some benchmark and noticed that this is better for performance reasons?
Or is there some other aspect I'm missing?

In theory `channel`s could be more efficient but they could maybe run out of memory in case of high traffic loads.
Was this your concern?

Thanks in advance!

**Comment by kpcyrd** [comment]
hey! :) I don't remember why I did it this way, it was one of my first Rust projects. Apparently in 2019 I briefly switched to `crossbeam_channel::bounded` (possibly because of https://github.com/rust-lang/rust/issues/39364), but I don't remember why I switched to sync_channel afterwards.
